cs/src/core/RuntimeSerializer.cs

```diff
@@ -0,0 +1,246 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Bond
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Linq;
+    using Bond.IO;
+    using Bond.IO.Safe;
+    using Bond.Protocols;
+
+    /// <summary>
+    /// Serializes <see cref="RuntimeObject"/>s to a protocol writer using only a runtime schema.
+    /// </summary>
+    public static class RuntimeSerialize
+    {
+        /// <summary>
+        /// Serializes a <see cref="RuntimeObject"/> to a protocol writer using only a runtime schema.
+        /// </summary>
+        /// <param name="writer">The protocol writer to write the object data to.</param>
+        /// <param name="schema">The schema to use when writing the object data.</param>
+        /// <param name="obj">The object to serialize.</param>
+        public static void To(IProtocolWriter writer, RuntimeSchema schema, RuntimeObject obj)
+        {
+            new RuntimeSerializer(schema).Serialize(writer, obj);
+        }
+    }
+
+    /// <summary>
+    /// Serializes <see cref="RuntimeObject"/>s to a protocol writer using only a runtime schema.
+    /// </summary>
+    public class RuntimeSerializer
```

Comment:
A better way would be to implement [IParser](https://github.com/Microsoft/bond/blob/1da75c2ffa91d8a20a22ee2cb9714aef28217872/cs/src/core/expressions/IParser.cs#L114-L118) for `RuntimeObject`. It would be conceptually similar to [`ObjectParser`](https://github.com/Microsoft/bond/blob/1da75c2ffa91d8a20a22ee2cb9714aef28217872/cs/src/core/expressions/ObjectParser.cs#L14).